department-of-veterans-affairs/caseflow
# Al/appeals 45664

Resolves [APPEALS-45664](https://jira.devops.va.gov/browse/APPEALS-45664)

# Description
Please explain the changes you made here.
1.  Added new `saved_searches_controller.rb` and its actions. `index, show, delete and create`.
2.  Added test for the controller -> `saved_searches_controller_spec.rb`
3.  Added Serializer for saved_search
4. Added two scopes in model to get the data from the backend. 
5. added a method to validate if its a VHA_ADMIN
6. Added template files for index and show. 


## Acceptance Criteria
- [ ] Handle saving of new saved search 
- [ ] Handle deletion of saved search 
- [ ]  Validation to ensure user can only delete their own Saved Search, not other users'
- [ ] Make index to return all Saved Searches 
- [ ] Make action to return saved searches for a specific user 
- [ ] Need validation to ensure User is a VHA admin

## Testing Plan

In Terminal: 
```
bundle exec rails runner scripts/run_seed_data_file.rb VhaGenerateTaskReport
```
1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2. You can use postman to test the functionality. 
3. login to postman, copy set-cookie from the browser and paste it in POSTMAN.
4. access -> GET: localhost:3000/saved_searches.json -> this should return all the saved search for all user. 
5. GET: localhost:3000/saved_searches/:id -> this should only return single saved search based on the id. 
6. POST: localhost:3000/saved_searches/ -> this should create a new saved search
7. DELETE: localhost:3000/saved_searches/:id.  > Thi should delete a saved search for that user.

Tester can use Postman using this [link](https://titin6.postman.co/workspace/Titin-Workspace~238b6a73-417b-47d7-ad7e-777ffc6708d1/folder/34775188-2b7de467-8f4f-4ba7-8622-07d50cee3014?action=share&source=copy-link&creator=34775188&ctx=documentation) with pre-saved URL endpoint criteria to ease help with testing. Also, need to add `X-CSRF-Token`


In order to make an API call tester needs to capture `_caseflow_session` from the Chrome browser after logging in through the frontend with the type of user they want to test the endpoint with e.g vha admin


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
